payjoin-cli/src/app/v2/mod.rs

```diff
@@ -500,8 +520,28 @@ impl App {
             match res {
                 Ok(OptionalTransitionOutcome::Progress(psbt)) => {
                     println!("Proposal received. Processing...");
-                    self.process_pj_response(psbt)?;
-                    return Ok(());
+                    let proposal_session = SendSession::ProposalReceived(psbt);
+                    let transition = proposal_session.process_proposal(|psbt| {
+                        match self.process_pj_response(psbt) {
+                            Ok(_) => Ok(()),
+                            Err(e) => {
+                                tracing::debug!("PSBT processing error: {:?}", e);
+                                Err(ImplementationError::from("Payjoin processing failed"))
+                            }
+                        }
+                    });
+
+                    match transition.save(persister) {
+                        Ok(OptionalTransitionOutcome::Progress(_)) => {
+                            return Ok(());
+                        }
+                        Ok(OptionalTransitionOutcome::Stasis(_)) => {
+                            return Err(anyhow!("Unexpected state after processing proposal"));
+                        }
+                        Err(e) => {
+                            return Err(anyhow!("Failed to save session transition: {}", e));
+                        }
+                    }
```

Comment:
What problem are you looking to solve with this change? IIRC the session is already closed at this point and should not be persisting more session event https://github.com/payjoin/rust-payjoin/blob/7ac750c14cbe5e4bc738592b48421621ebf9994a/payjoin/src/core/send/v2/mod.rs#L525